WooCommerce/Classes/ViewRelated/Products/AI/ProductSharingMessageGenerationView.swift

```diff
@@ -0,0 +1,143 @@
+    init(viewModel: ProductSharingMessageGenerationViewModel,
+         onShareMessage: @escaping (String) -> Void) {
+        super.init(rootView: ProductSharingMessageGenerationView(viewModel: viewModel,
+                                                                 onShareMessage: onShareMessage))
+    }
+
+    @available(*, unavailable)
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+/// View for generating product sharing message with AI.
+struct ProductSharingMessageGenerationView: View {
+    @ObservedObject private var viewModel: ProductSharingMessageGenerationViewModel
+    private let onShareMessage: (String) -> Void
+
+    init(viewModel: ProductSharingMessageGenerationViewModel,
+         onShareMessage: @escaping (String) -> Void) {
+        self.viewModel = viewModel
+        self.onShareMessage = onShareMessage
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: Constants.defaultSpacing) {
+
+            // View title
+            Text(viewModel.viewTitle)
+                .headlineStyle()
+
+            Divider()
+
+            // Generated message text field
+            ZStack(alignment: .topLeading) {
+                TextEditor(text: $viewModel.messageContent)
+                    .bodyStyle()
+                    .foregroundColor(.secondary)
+                    .background(viewModel.generationInProgress ? Color(uiColor: .buttonDisabledBackground) : .clear)
+                    .padding(insets: Constants.messageContentInsets)
+                    .overlay(
+                        RoundedRectangle(cornerRadius: Constants.cornerRadius).stroke(Color(.separator))
+                    )
+
+                // Placeholder text
+                Text(Localization.placeholder)
+                    .foregroundColor(Color(.placeholderText))
+                    .bodyStyle()
+                    .padding(insets: Constants.placeholderInsets)
+                    // Allows gestures to pass through to the `TextEditor`.
+                    .allowsHitTesting(false)
+                    .renderedIf(viewModel.messageContent.isEmpty)
+            }
+            .renderedIf(viewModel.generationInProgress == false)
+
+            // Skeleton view for loading state
+            Text(Constants.dummyText)
+                .secondaryBodyStyle()
+                .padding(Constants.placeholderInsets)
+                .background(RoundedRectangle(cornerRadius: Constants.cornerRadius).stroke(Color(uiColor: .secondarySystemFill)))
+                .redacted(reason: .placeholder)
+                .renderedIf(viewModel.generationInProgress)
+
+            // Error message
+            viewModel.errorMessage.map { message in
+                Text(message).errorStyle()
+            }
+
+            // Action button to generate message
+            Button(action: {
+                Task { @MainActor in
```

Comment:
Nit: Do we need to mention `@MainActor` here considering that `generateShareMessage` is already marked as `MainActor`?